SCClassLibrary/Common/Math/Rational.sc

```diff
@@ -7,23 +7,25 @@ Rational : Number {
 
 	reduce {
 		var d;
-		if (this.numerator.isKindOf(Number)) {
-			if (this.numerator.isKindOf(Rational) || this.denominator.isKindOf(Rational)){
-				^(numerator.asRational / denominator.asRational)
-			};
-			if (this.numerator.frac == 0 && this.denominator.frac == 0) {
-				d = this.factor;
-				numerator   = ((this.numerator/d).abs * d.sign).round;
-				denominator = (this.denominator/d).abs.round;
-				if (denominator == 0) {"Rational has zero denominator".error};
-			} {
-				^(this.numerator / this.denominator).asRational
-			}
-		} {
-			if (this.numerator.isKindOf(String)) {
-				^this.numerator.asRational
-			}
-		}
+        if (this.numerator.isKindOf(Number)) {
```

Comment:
does this method return a new instance or not? Sometimes it does it either way, it seems - better keep just one. Maybe best return a new one.